Updates

JoshData opened the pull request:
Hi all. I'm working on an update to https://uslaw.link/. Here are some updates to citation. The first few commits in this PR are just about updating packages with security issues.  The last few add some minor new functionality.

JoshData committed:
ran 'npm audit fix'

JoshData committed:
update gulp

Ran 'npm audit fix --force' and then fixed the gulpfile.

JoshData committed:
re-ran gulp

JoshData committed:
add package-lock.json

JoshData committed:
various tweaks to the linker classes

* Update GPO (govinfo) and uscode.house.gov links.
* Remove 'beta' from code.dccouncil.us links.
* Remove legislink.org linker because the site is being discontinued.
* Disable the LoC links for Statutes at Large volumes because it's not a deep link.
* Other tweaks.

JoshData committed:
add type_name to output

JoshData committed:
add Citation.fromId and fromId to citators to facilitate creating permalinks

JoshData committed:
re-ran gulp

divergentdave commented on links/libraryofcongress.js:
```diff
@@ -8,6 +8,7 @@ module.exports = {
   authoritative: true,
 
   citations: {
+    /*
```
What's the disposition of this function?

JoshData replied:
The function remains a somewhat helpful example of how we might generate links to a Library of Congress resource for the Statutes at Large, but because it links to a main landing page and not a particular volume or page of the SaL, I commented it out because it's not very helpful.

divergentdave reviewed (commented):
Looks pretty good, thanks for working on this! (I haven't kicked the tires on it, as I'm writing this review from a bus) I think we might benefit from some tests to compare the output of fromId to the existing find method. The test could find a citation from a fixed string, take the id from that object, pass it to fromId, and then do a deep comparison of the two objects. They should be identical, save for the fields relating to the text searched by find. Thanks again!

divergentdave committed:
Fix link tests

JoshData committed:
support non-integer numbers in PubL citations

JoshData committed:
Add tests for Citation.fromId, 'type'/'type_name' fields, historical anomalous Pub.L. numbers

JoshData commented:
Tests added! :)

JoshData commented:
It looks like nodeunit is permanently broken --- I can't get it to install on Travis, so the build is failing.

But it works fine locally and tests are passing. Everyone OK to merge?

divergentdave commented:
I think the build is failing because `.travis.yml` specifies a very old version of node, and ejs's postinstall script is too new for it.

JoshData committed:
Remove old node version specified in travis.yml

The ejs package required by nodeunit failed to install on this old version. .travis.yml is revised to use the latest stable node version per https://docs.travis-ci.com/user/languages/javascript-with-nodejs/.

divergentdave commented:
Oh dear, it's using v0.10 now, the default is slightly older!

JoshData commented:
Nice catch! Ok finally got it working.

Heydude13 reviewed (changes requested):
![Uploading Content Restrictions-2015_H1(1).csv.pdf …]()
